### ISSUE TITLE
rolling forecast

### ISSUE THREAD

**Issue by sonichi** [microsoft/FLAML#750]
Allow a time series forecasting model to predict beyond the horizon after new data arrive, without refitting the model. For example,
```python
# predict for the first time, assuming the time horizon = 1 hour. The last timestamp in the training data is 7am.
y_pred_8am = automl.predict(X_pred_8am)  # X_pred_8am's timestamp = 8am
# after y_true_8am is observed
X_roll, y_roll = X_pred_8am, y_pred_8am
y_pred_9am = automl.predict(X_pred_9am, X_roll, y_roll)
# after y_true_9am is observed
X_roll = concat(X_roll, X_pred_9am)
y_roll = concat(y_roll, y_pred_9am)
y_pred_10am = automl.predict(X_pred_10am, X_roll, y_roll)
...
```

@int-chaos @egor-bryzgalov @markharley @slhuang @qingyun-wu @luigif2000 @MichalChromcak Does this feature make sense?

**Comment by Papa-Panda** [comment]
Yes. We need to train the model and then use the model without refitting
for a while.

**Comment by sonichi** [comment]
> Yes. We need to train the model and then use the model without refitting for a while.

Thanks. Just to confirm, you need to use the model beyond the forecast horizon, right? Roughly how long is the forecast horizon and how long do you need to keep using the model before refitting?

**Comment by Papa-Panda** [comment]
Forecasting horizon is from 1 day to 4 weeks.

Another scenario is forecast 1 hour but for this we do not need to keep
using model before refitting.

**Comment by sonichi** [comment]
> Forecasting horizon is from 1 day to 4 weeks. Another scenario is forecast 1 hour but for this we do not need to keep using model before refitting.

Thanks. For the scenario where the forecasting horizon is 1 day, how long would you keep using the same model before refitting?

**Comment by Papa-Panda** [comment]
I will keep using the model for a week. So model training is weekly.
